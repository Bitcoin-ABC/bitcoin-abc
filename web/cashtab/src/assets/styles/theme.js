export const theme = {
    primary: '#00ABE7',
    brandSecondary: '#CD0BC3',
    contrast: '#fff',
    app: {
        sidebars: `url("/cashtab_bg.png")`,
        background: '#fbfbfd',
    },
    wallet: {
        background: '#fff',
        text: {
            primary: '#273498',
            secondary: '#273498',
        },
        switch: {
            activeCash: {
                shadow:
                    'inset 8px 8px 16px #0074C2, inset -8px -8px 16px #273498',
            },
            activeToken: {
                background: '#CD0BC3',
                shadow:
                    'inset 5px 5px 11px #FF21D0, inset -5px -5px 11px #CD0BC3',
            },
            inactive: {
                background: 'linear-gradient(145deg, #eeeeee, #c8c8c8)',
            },
        },
        borders: { color: '#e2e2e2' },
        shadow: 'rgba(0, 0, 0, 1)',
    },
    tokenListItem: {
        background: '#ffffff',
        color: '',
        boxShadow:
            'rgba(0, 0, 0, 0.01) 0px 0px 1px, rgba(0, 0, 0, 0.04) 0px 4px 8px,rgba(0, 0, 0, 0.04) 0px 16px 24px, rgba(0, 0, 0, 0.01) 0px 24px 32px',
        border: '#e9eaed',
        hoverBorder: '#231F20',
    },
    footer: {
        background: '#fff',
        navIconInactive: '#949494',
    },
    forms: {
        error: '#FF21D0',
        border: '#e7edf3',
        text: '#001137',
        addonBackground: '#f4f4f4',
        addonForeground: '#3e3f42',
        selectionBackground: '#fff',
    },
    icons: { outlined: '#273498' },
    modals: {
        buttons: { background: '#fff' },
    },
    settings: { delete: '#CD0BC3' },
    qr: {
        copyBorderCash: '#00ABE7',
        copyBorderToken: '#FF21D0',
        background: '#fff',
        token: '#231F20',
        shadow:
            'rgba(0, 0, 0, 0.01) 0px 0px 1px, rgba(0, 0, 0, 0.04) 0px 4px 8px, rgba(0, 0, 0, 0.04) 0px 16px 24px, rgba(0, 0, 0, 0.01) 0px 24px 32px',
    },
    buttons: {
        primary: {
            backgroundImage:
                'linear-gradient(270deg, #0074C2 0%, #273498 100%)',
            color: '#fff',
            hoverShadow: '0px 3px 10px -5px rgba(0, 0, 0, 0.75)',
        },
        secondary: {
            background: '#e9eaed',
            color: '#444',
            hoverShadow: '0px 3px 10px -5px rgba(0, 0, 0, 0.75)',
        },
    },
    collapses: {
        background: '#fbfcfd',
        border: '#eaedf3',
        color: '#3e3f42',
    },
};
