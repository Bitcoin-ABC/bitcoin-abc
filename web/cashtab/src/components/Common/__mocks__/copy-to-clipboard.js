const copy = jest.fn();
export default copy;
