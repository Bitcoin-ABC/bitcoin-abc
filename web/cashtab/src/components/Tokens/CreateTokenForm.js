import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AntdFormWrapper } from '@components/Common/EnhancedInputs';
import { TokenCollapse } from '@components/Common/StyledCollapse';
import { currency } from '@components/Common/Ticker.js';
import { WalletContext } from '@utils/context';
import {
    isValidTokenName,
    isValidTokenTicker,
    isValidTokenDecimals,
    isValidTokenInitialQty,
    isValidTokenDocumentUrl,
} from '@utils/validation';
import { PlusSquareOutlined } from '@ant-design/icons';
import { SmartButton } from '@components/Common/PrimaryButton';
import { Collapse, Form, Input, Modal, notification } from 'antd';
const { Panel } = Collapse;
import Paragraph from 'antd/lib/typography/Paragraph';
import { TokenParamLabel } from '@components/Common/Atoms';
import { TokenReceivedNotificationIcon } from '@components/Common/CustomIcons';

const CreateTokenForm = ({
    BCH,
    getRestUrl,
    createToken,
    disabled,
    passLoadingStatus,
}) => {
    const { wallet } = React.useContext(WalletContext);

    // New Token Name
    const [newTokenName, setNewTokenName] = useState('');
    const [newTokenNameIsValid, setNewTokenNameIsValid] = useState(null);
    const handleNewTokenNameInput = e => {
        const { value } = e.target;
        // validation
        setNewTokenNameIsValid(isValidTokenName(value));
        setNewTokenName(value);
    };

    // New Token Ticker
    const [newTokenTicker, setNewTokenTicker] = useState('');
    const [newTokenTickerIsValid, setNewTokenTickerIsValid] = useState(null);
    const handleNewTokenTickerInput = e => {
        const { value } = e.target;
        // validation
        setNewTokenTickerIsValid(isValidTokenTicker(value));
        setNewTokenTicker(value);
    };

    // New Token Decimals
    const [newTokenDecimals, setNewTokenDecimals] = useState(0);
    const [newTokenDecimalsIsValid, setNewTokenDecimalsIsValid] = useState(
        true,
    );
    const handleNewTokenDecimalsInput = e => {
        const { value } = e.target;
        // validation
        setNewTokenDecimalsIsValid(isValidTokenDecimals(value));
        // Also validate the supply here if it has not yet been set
        if (newTokenInitialQtyIsValid !== null) {
            setNewTokenInitialQtyIsValid(
                isValidTokenInitialQty(value, newTokenDecimals),
            );
        }

        setNewTokenDecimals(value);
    };

    // New Token Initial Quantity
    const [newTokenInitialQty, setNewTokenInitialQty] = useState('');
    const [newTokenInitialQtyIsValid, setNewTokenInitialQtyIsValid] = useState(
        null,
    );
    const handleNewTokenInitialQtyInput = e => {
        const { value } = e.target;
        // validation
        setNewTokenInitialQtyIsValid(
            isValidTokenInitialQty(value, newTokenDecimals),
        );
        setNewTokenInitialQty(value);
    };
    // New Token document URL
    const [newTokenDocumentUrl, setNewTokenDocumentUrl] = useState('');
    // Start with this as true, field is not required
    const [
        newTokenDocumentUrlIsValid,
        setNewTokenDocumentUrlIsValid,
    ] = useState(true);

    const handleNewTokenDocumentUrlInput = e => {
        const { value } = e.target;
        // validation
        setNewTokenDocumentUrlIsValid(isValidTokenDocumentUrl(value));
        setNewTokenDocumentUrl(value);
    };

    // New Token fixed supply
    // Only allow creation of fixed supply tokens until Minting support is added

    // New Token document hash
    // Do not include this; questionable value to casual users and requires significant complication

    // Only enable CreateToken button if all form entries are valid
    let tokenGenesisDataIsValid =
        newTokenNameIsValid &&
        newTokenTickerIsValid &&
        newTokenDecimalsIsValid &&
        newTokenInitialQtyIsValid &&
        newTokenDocumentUrlIsValid;

    // Modal settings
    const [showConfirmCreateToken, setShowConfirmCreateToken] = useState(false);

    const createPreviewedToken = async () => {
        passLoadingStatus(true);
        // If data is for some reason not valid here, bail out
        if (!tokenGenesisDataIsValid) {
            return;
        }

        // data must be valid and user reviewed to get here
        const configObj = {
            name: newTokenName,
            ticker: newTokenTicker,
            documentUrl:
                newTokenDocumentUrl === ''
                    ? 'https://cashtabapp.com/'
                    : newTokenDocumentUrl,
            decimals: newTokenDecimals,
            initialQty: newTokenInitialQty,
            documentHash: '',
        };

        // create token with data in state fields
        try {
            const link = await createToken(
                BCH,
                wallet,
                currency.defaultFee,
                configObj,
            );

            notification.success({
                message: 'Success',
                description: (
                    <a href={link} target="_blank" rel="noopener noreferrer">
                        <Paragraph>
                            Token created! Click to view in block explorer.
                        </Paragraph>
                    </a>
                ),
                icon: <TokenReceivedNotificationIcon />,
                style: { width: '100%' },
            });
        } catch (e) {
            // Set loading to false here as well, as balance may not change depending on where error occured in try loop
            passLoadingStatus(false);
            let message;

            if (!e.error && !e.message) {
                message = `Transaction failed: no response from ${getRestUrl()}.`;
            } else if (
                /Could not communicate with full node or other external service/.test(
                    e.error,
                )
            ) {
                message = 'Could not communicate with API. Please try again.';
            } else if (
                e.error &&
                e.error.includes(
                    'too-long-mempool-chain, too many unconfirmed ancestors [limit: 50] (code 64)',
                )
            ) {
                message = `The ${currency.ticker} you are trying to send has too many unconfirmed ancestors to send (limit 50). Sending will be possible after a block confirmation. Try again in about 10 minutes.`;
            } else {
                message = e.message || e.error || JSON.stringify(e);
            }

            notification.error({
                message: 'Error',
                description: message,
                duration: 5,
            });
            console.error(e);
        }
        // Hide the modal
        setShowConfirmCreateToken(false);
        // Stop spinner
        passLoadingStatus(false);
    };
    return (
        <>
            <Modal
                title={`Please review and confirm your token settings.`}
                visible={showConfirmCreateToken}
                onOk={createPreviewedToken}
                onCancel={() => setShowConfirmCreateToken(false)}
            >
                <TokenParamLabel>Name:</TokenParamLabel> {newTokenName}
                <br />
                <TokenParamLabel>Ticker:</TokenParamLabel> {newTokenTicker}
                <br />
                <TokenParamLabel>Decimals:</TokenParamLabel> {newTokenDecimals}
                <br />
                <TokenParamLabel>Supply:</TokenParamLabel> {newTokenInitialQty}
                <br />
                <TokenParamLabel>Document URL:</TokenParamLabel>{' '}
                {newTokenDocumentUrl === ''
                    ? 'https://cashtabapp.com/'
                    : newTokenDocumentUrl}
                <br />
            </Modal>
            <>
                <TokenCollapse
                    collapsible={disabled ? 'disabled' : true}
                    disabled={disabled}
                    style={{
                        marginBottom: '24px',
                    }}
                >
                    <Panel header="Create Token" key="1">
                        <AntdFormWrapper>
                            <Form
                                size="small"
                                style={{
                                    width: 'auto',
                                }}
                            >
                                <Form.Item
                                    validateStatus={
                                        newTokenNameIsValid === null ||
                                        newTokenNameIsValid
                                            ? ''
                                            : 'error'
                                    }
                                    help={
                                        newTokenNameIsValid === null ||
                                        newTokenNameIsValid
                                            ? ''
                                            : 'Token name must be a string between 1 and 68 characters long'
                                    }
                                >
                                    <Input
                                        addonBefore="Name"
                                        placeholder="Enter a name for your token"
                                        name="newTokenName"
                                        value={newTokenName}
                                        onChange={e =>
                                            handleNewTokenNameInput(e)
                                        }
                                    />
                                </Form.Item>
                                <Form.Item
                                    validateStatus={
                                        newTokenTickerIsValid === null ||
                                        newTokenTickerIsValid
                                            ? ''
                                            : 'error'
                                    }
                                    help={
                                        newTokenTickerIsValid === null ||
                                        newTokenTickerIsValid
                                            ? ''
                                            : 'Ticker must be a string between 1 and 12 characters long'
                                    }
                                >
                                    <Input
                                        addonBefore="Ticker"
                                        placeholder="Enter a ticker for your token"
                                        name="newTokenTicker"
                                        value={newTokenTicker}
                                        onChange={e =>
                                            handleNewTokenTickerInput(e)
                                        }
                                    />
                                </Form.Item>
                                <Form.Item
                                    validateStatus={
                                        newTokenDecimalsIsValid === null ||
                                        newTokenDecimalsIsValid
                                            ? ''
                                            : 'error'
                                    }
                                    help={
                                        newTokenDecimalsIsValid === null ||
                                        newTokenDecimalsIsValid
                                            ? ''
                                            : 'Token decimals must be an integer between 0 and 9'
                                    }
                                >
                                    <Input
                                        addonBefore="Decimals"
                                        placeholder="Enter number of decimal places"
                                        name="newTokenDecimals"
                                        type="number"
                                        value={newTokenDecimals}
                                        onChange={e =>
                                            handleNewTokenDecimalsInput(e)
                                        }
                                    />
                                </Form.Item>
                                <Form.Item
                                    validateStatus={
                                        newTokenInitialQtyIsValid === null ||
                                        newTokenInitialQtyIsValid
                                            ? ''
                                            : 'error'
                                    }
                                    help={
                                        newTokenInitialQtyIsValid === null ||
                                        newTokenInitialQtyIsValid
                                            ? ''
                                            : 'Token supply must be greater than 0 and less than 100,000,000,000. Token supply decimal places cannot exceed token decimal places.'
                                    }
                                >
                                    <Input
                                        addonBefore="Supply"
                                        placeholder="Enter the fixed supply of your token"
                                        name="newTokenInitialQty"
                                        type="number"
                                        value={newTokenInitialQty}
                                        onChange={e =>
                                            handleNewTokenInitialQtyInput(e)
                                        }
                                    />
                                </Form.Item>
                                <Form.Item
                                    validateStatus={
                                        newTokenDocumentUrlIsValid === null ||
                                        newTokenDocumentUrlIsValid
                                            ? ''
                                            : 'error'
                                    }
                                    help={
                                        newTokenDocumentUrlIsValid === null ||
                                        newTokenDocumentUrlIsValid
                                            ? ''
                                            : 'Document URL cannot exceed 68 characters'
                                    }
                                >
                                    <Input
                                        addonBefore="Document URL"
                                        placeholder="Enter a website for your token"
                                        name="newTokenDocumentUrl"
                                        value={newTokenDocumentUrl}
                                        onChange={e =>
                                            handleNewTokenDocumentUrlInput(e)
                                        }
                                    />
                                </Form.Item>
                            </Form>
                        </AntdFormWrapper>
                        <SmartButton
                            onClick={() => setShowConfirmCreateToken(true)}
                            disabled={!tokenGenesisDataIsValid}
                        >
                            <PlusSquareOutlined />
                            &nbsp;Create Token
                        </SmartButton>
                    </Panel>
                </TokenCollapse>
            </>
        </>
    );
};

/*
passLoadingStatus must receive a default prop that is a function
in order to pass the rendering unit test in CreateTokenForm.test.js

status => {console.log(status)} is an arbitrary stub function
*/

CreateTokenForm.defaultProps = {
    passLoadingStatus: status => {
        console.log(status);
    },
};

CreateTokenForm.propTypes = {
    BCH: PropTypes.object,
    getRestUrl: PropTypes.func,
    createToken: PropTypes.func,
    disabled: PropTypes.bool,
    passLoadingStatus: PropTypes.func,
};

export default CreateTokenForm;
