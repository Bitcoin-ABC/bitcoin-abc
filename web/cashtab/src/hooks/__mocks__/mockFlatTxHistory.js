export default [
    {
        address: 'bitcoincash:qqvcsnz9x9nu7vq35vmrkjc7hkfxhhs9nuv44zm0ed',
        height: 678992,
        txid:
            '5072cf021e0301284cf1cc1eb67dcb73fe8eb1f2a4fc3f8f3944d232e3946551',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 678992,
        txid:
            '82845d3c814d715b2c36aea0b076cc03815469a9c172c5bab7fc6a9f71b1906d',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 678616,
        txid:
            '7abd604770d183ad33d3431c93b29b324b82cd11a6cf0671b4edb7d44d9b2a40',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 678000,
        txid:
            '9897f172154bb8f27bbf4e0385926a535267f044cd11232a7efca91bece7dba6',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 677989,
        txid:
            'dee2ec72b0a22ceb99e67731de086a2545aaa9dc3fb4a8bdf9dddbae215d8f6c',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 677989,
        txid:
            'b854a48079e42612bed3b68cf62a90a021f20ae28e694dbe5dbfeb29ee8fea98',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 677989,
        txid:
            '9520af2513b54b1c47a8a520251d26f057ccbe986cb569682aa963bab626ba52',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 677989,
        txid:
            '72acfdb2cd6ab8cb44f07a6fd78ae4aa31c304ea3ff0206b50bef60ff953d94d',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 677752,
        txid:
            '8d47af8e46afc3cd908ddedf7f330ee849004d2464d65bd3b225518e924c9477',
    },
    {
        address: 'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 674993,
        txid:
            '089f2188d5771a7de0589def2b8d6c1a1f33f45b6de26d9a0ef32782f019ecf1',
    },
];
