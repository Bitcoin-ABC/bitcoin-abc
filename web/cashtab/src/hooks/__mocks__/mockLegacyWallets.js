export default {
    legacyAlpha: {
        mnemonic:
            'apart vacuum color cream drama kind foil history hurt alone ask census',
        name: 'MigrationTestAlpha',
        Path245: {
            cashAddress:
                'bitcoincash:qztqe8k4v8ckn8cvfxt5659nhd7dcyvxy54hkry298',
            slpAddress:
                'simpleledger:qztqe8k4v8ckn8cvfxt5659nhd7dcyvxy5evac32me',
            fundingWif: 'KwgNkyijAaxFr5XQdnaYyNMXVSZobgHzSoKKfWiC3Q7Xr4n7iYMG',
            fundingAddress:
                'simpleledger:qztqe8k4v8ckn8cvfxt5659nhd7dcyvxy5evac32me',
            legacyAddress: '1EgPUfBgU7ekho3EjtGze87dRADnUE8ojP',
        },
        Path145: {
            cashAddress:
                'bitcoincash:qq47pcxfn8n7w7jy86njd7pvgsv39l9f9v0lgx569z',
            slpAddress:
                'simpleledger:qq47pcxfn8n7w7jy86njd7pvgsv39l9f9vryrap6mu',
            fundingWif: 'L2xvTe6CdNxroR6pbdpGWNjAa55AZX5Wm59W5TXMuH31ihNJdDjt',
            fundingAddress:
                'simpleledger:qq47pcxfn8n7w7jy86njd7pvgsv39l9f9vryrap6mu',
            legacyAddress: '1511T3ynXKgCwXhFijCUWKuTfqbPxFV1AF',
        },
    },
    migratedLegacyAlpha: {
        mnemonic:
            'apart vacuum color cream drama kind foil history hurt alone ask census',
        name: 'MigrationTestAlpha',
        Path245: {
            cashAddress:
                'bitcoincash:qztqe8k4v8ckn8cvfxt5659nhd7dcyvxy54hkry298',
            slpAddress:
                'simpleledger:qztqe8k4v8ckn8cvfxt5659nhd7dcyvxy5evac32me',
            fundingWif: 'KwgNkyijAaxFr5XQdnaYyNMXVSZobgHzSoKKfWiC3Q7Xr4n7iYMG',
            fundingAddress:
                'simpleledger:qztqe8k4v8ckn8cvfxt5659nhd7dcyvxy5evac32me',
            legacyAddress: '1EgPUfBgU7ekho3EjtGze87dRADnUE8ojP',
        },
        Path145: {
            cashAddress:
                'bitcoincash:qq47pcxfn8n7w7jy86njd7pvgsv39l9f9v0lgx569z',
            slpAddress:
                'simpleledger:qq47pcxfn8n7w7jy86njd7pvgsv39l9f9vryrap6mu',
            fundingWif: 'L2xvTe6CdNxroR6pbdpGWNjAa55AZX5Wm59W5TXMuH31ihNJdDjt',
            fundingAddress:
                'simpleledger:qq47pcxfn8n7w7jy86njd7pvgsv39l9f9vryrap6mu',
            legacyAddress: '1511T3ynXKgCwXhFijCUWKuTfqbPxFV1AF',
        },
        Path1899: {
            cashAddress:
                'bitcoincash:qzagy47mvh6qxkvcn3acjnz73rkhkc6y7cptzgcqy6',
            slpAddress:
                'simpleledger:qzagy47mvh6qxkvcn3acjnz73rkhkc6y7cdsfndq6y',
            fundingWif: 'Kx4FiBMvKK1iXjFk5QTaAK6E4mDGPjmwDZ2HDKGUZpE4gCXMaPe9',
            fundingAddress:
                'simpleledger:qzagy47mvh6qxkvcn3acjnz73rkhkc6y7cdsfndq6y',
            legacyAddress: '1J1Aq5tAAYxZgSDRo8soKM2Rb41z3xrYpm',
        },
    },
};
