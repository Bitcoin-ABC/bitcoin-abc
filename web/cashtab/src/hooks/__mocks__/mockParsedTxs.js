// Expected result of applying parseTxData to mockTxDataWityhPassthrough[0]
export const mockSentCashTx = [
    {
        amountReceived: 0,
        amountSent: 0.000042,
        blocktime: 1614380741,
        confirmations: 2721,
        destinationAddress:
            'bitcoincash:qphpmfj0qn7znklqhrfn5dq7qh36l3vxavu346vqcl',
        height: 674993,
        outgoingTx: true,
        tokenTx: false,
        txid:
            '089f2188d5771a7de0589def2b8d6c1a1f33f45b6de26d9a0ef32782f019ecf1',
    },
];

export const mockReceivedCashTx = [
    {
        amountReceived: 3,
        amountSent: 0,
        blocktime: 1612567121,
        confirmations: 5637,
        destinationAddress:
            'bitcoincash:qpmytrdsakt0axrrlswvaj069nat3p9s7ct4lsf8k9',
        height: 672077,
        outgoingTx: false,
        tokenTx: false,
        txid:
            '42d39fbe068a40fe691f987b22fdf04b80f94d71d2fec20a58125e7b1a06d2a9',
    },
];

export const mockSentTokenTx = [
    {
        amountReceived: 0,
        amountSent: 0.00000546,
        blocktime: 1614027278,
        confirmations: 3270,
        destinationAddress:
            'bitcoincash:qzj5zu6fgg8v2we82gh76xnrk9njcregluzgaztm45',
        height: 674444,
        outgoingTx: true,
        tokenTx: true,
        txid:
            'ffe3a7500dbcc98021ad581c98d9947054d1950a7f3416664715066d3d20ad72',
    },
];
export const mockReceivedTokenTx = [
    {
        amountReceived: 0.00000546,
        amountSent: 0,
        blocktime: 1613859311,
        confirmations: 3571,
        destinationAddress:
            'bitcoincash:qpmytrdsakt0axrrlswvaj069nat3p9s7ct4lsf8k9',
        height: 674143,
        outgoingTx: false,
        tokenTx: true,
        txid:
            '618d0dd8c0c5fa5a34c6515c865dd72bb76f8311cd6ee9aef153bab20dabc0e6',
    },
];
