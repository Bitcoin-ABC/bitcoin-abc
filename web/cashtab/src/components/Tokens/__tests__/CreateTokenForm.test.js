import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from '@assets/styles/theme';
import CreateTokenForm from '@components/Tokens/CreateTokenForm';
import BCHJS from '@psf/bch-js';
import useBCH from '@hooks/useBCH';
import { walletWithBalancesAndTokensWithCorrectState } from '../../Wallet/__mocks__/walletAndBalancesMock';

let useContextMock;
let realUseContext;

beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();

    // Mock method not implemented in JSDOM
    // See reference at https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation(query => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
});

afterEach(() => {
    React.useContext = realUseContext;
});

test('Wallet with BCH balances and tokens and state field', () => {
    useContextMock.mockReturnValue(walletWithBalancesAndTokensWithCorrectState);

    const testBCH = new BCHJS();
    const { getRestUrl, createToken } = useBCH();
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <CreateTokenForm
                BCH={testBCH}
                getRestUrl={getRestUrl}
                createToken={createToken}
                disabled={
                    walletWithBalancesAndTokensWithCorrectState.wallet.state
                        .balances.totalBalanceInSatoshis < 546
                }
            />
        </ThemeProvider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
