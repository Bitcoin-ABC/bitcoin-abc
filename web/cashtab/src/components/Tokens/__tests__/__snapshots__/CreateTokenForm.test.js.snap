// Jest Snapshot v1, https://goo.gl/fbAQLP @generated

exports[`Wallet with BCH balances and tokens and state field 1`] = `
<div
  className="ant-collapse ant-collapse-icon-position-left sc-gqjmRU fLjHet"
  role={null}
  style={
    Object {
      "marginBottom": "24px",
    }
  }
>
  <div
    className="ant-collapse-item"
  >
    <div
      aria-expanded={false}
      className="ant-collapse-header"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        aria-label="right"
        className="anticon anticon-right ant-collapse-arrow"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
      Create Token
    </div>
  </div>
</div>
`;
